src/VisualStudio/Core/Def/TableDataSource/DiagnosticTableItem.cs

```diff
@@ -98,24 +98,21 @@ public bool Equals(DiagnosticData left, DiagnosticData right)
                 // location-less or project level diagnostic:
                 if (leftLocation == null ||
                     rightLocation == null ||
-                    leftLocation.OriginalFilePath == null ||
-                    rightLocation.OriginalFilePath == null ||
+                    leftLocation?.OriginalFileSpan?.Path == null ||
+                    rightLocation?.OriginalFileSpan?.Path == null ||
```

Comment:
this is also deeply upsetting.  i'm just not at all clear what the semantics of this stuff is supposed to be or why there's so many unique different ways in the existing code that locations are compared.